main.go

```diff
@@ -62,12 +71,53 @@ func main() {
 	}
 
 	// todo use token to communicate with snyk api
-	apiClient := NewApiClient(snykAPIToken)
+	client := NewApiClient(os.Getenv("ESTAFETTE_CREDENTIALS_SNYK_API_TOKEN"))
+
+	// The Organization ID will be fixed
+	orgID := os.Getenv("ESTAFETTE_ORGANIZATION_ID")
+	projectName := os.Getenv("ESTAFETTE_PROJECT_NAME")
+	targetScoreStr := os.Getenv("ESTAFETTE_TARGET_SCORE")
+	targetScore, err := strconv.ParseFloat(targetScoreStr, 64)
+
+	if err != nil {
+		log.Fatal().Err(err).Msg("Target Score provided is not a float variable")
+	}
+
+	projects, err := client.OrganizationProjects(ctx, orgID)
 
-	status, err := apiClient.GetStatus(ctx, *gitSource, *gitOwner, *gitName)
 	if err != nil {
-		log.Fatal().Err(err).Msg("Failed retrieving status from Snyk API")
+		log.Fatal().Msgf(err.Error())
+	}
+
+	var vulnerabilities snyk.Vulnerabilities
+
+	for _, project := range projects {
```

Comment:
Doesn't a project map 1:1 on a git repository? If that's the case this seems wildly inefficient and you would be better of trying to retrieve the issues for the current project this runs for and swallow any 404 errors.

And if a project is something wider (like a github organization) then the list of vulnerabilities retrieved doesn't just apply to the current pipeline, so you would still have to filter vulnerabilities relevant to the current pipeline.

Reply:
Not exactly. A repository can generate multiple projects in Snyk (for example, for different branches). I think we will only use master branches, even then, the repo name coming from Estafette is not exactly the same as in Snyk. One example is the following:

Git repo: xivart/sw-bookingidentity-data-service
Snyk project: xivart/sw-bookingidentity-data-service:pom.xml

Snyk holds an ID for each project, but Estafette doesn't know about it, so the solution for now is to iterate the projects and search for the right one. Unless we find a way of storing Snyk id for each repo somewhere, only then we can query for the right project without filtering the results I guess. =/

Reply:
I agree with you, this isn't the best case scenario and could even lead to errors while searching for the right project. The Snyk id looks like this: `3651ef26-641b-4ec9-8e47-7812c99e438a`

The only thing I can think of is manually configure this id on the manifest for each project, which is obviously a lot of work. If you guys have any other idea for automatically attaching this Snyk ID to each repo, I'd love to hear. 😄

Reply:
Sounds like quite a silly api. I guess we need to tag the projects with the repo source, owner and name (or see if those tags are already there) and use those when listing projects to filter it down to just the relevant project(s).

Reply:
See https://snyk.docs.apiary.io/#reference/projects/all-projects/list-all-projects

Reply:
Yes, it makes sense. I was using this apiary documentation as reference. 
I checked the response and all projects have no tags so far. 

`"tags": []`

We can try to run a script and tag all the projects on Snyk. But then every time we create a new project that has to be done. I will ask Jubba if by any chance we can configure that on the Snyk-BitBucket integration.